ksql-engine/src/main/java/io/confluent/ksql/util/GenericRowValueTypeEnforcer.java

```diff
@@ -37,22 +37,24 @@ public Object enforceFieldType(final int index, final Object value) {
   }
 
   private Object enforceFieldType(Schema schema, final Object value) {
-    if (schema == Schema.FLOAT64_SCHEMA) {
-      return enforceDouble(value);
-    } else if (schema == Schema.INT64_SCHEMA) {
-      return enforceLong(value);
-    } else if (schema == Schema.INT32_SCHEMA) {
-      return enforceInteger(value);
-    } else if (schema == Schema.STRING_SCHEMA) {
-      return enforceString(value);
-    } else if (schema == Schema.BOOLEAN_SCHEMA) {
-      return enforceBoolean(value);
-    } else if (schema.type() == Schema.Type.ARRAY) {
-      return value;
-    } else if (schema.type() == Schema.Type.MAP) {
-      return value;
-    } else {
-      throw new KsqlException("Type is not supported: " + schema);
+
+    switch (schema.type()) {
```

Comment:
this could also be push down do an enum on Schema.INT32.enforceInteger() etc - then you dont need this class

Reply:
@hjafarpour  - I still dont see more OO modelling in the latest commit...

Reply:
Oops, I forgot to submit my comments yesterday. @bluemonk3y as I mentioned above we use Schema class from Connect package so we don't have control on the  class implementation.

Reply:
We can create our own class to do this - i.e., one that wraps the Schema

Reply:
+1 - the code with have in here needs a serious refactor